src/integration_test/java/com/klodnicki/Bike/v2/rest/controller/RentBikeControllerIntegrationTest.java

```diff
@@ -0,0 +1,226 @@
+                    assertEquals(bike.getSerialNumber(), responseDTO.getBikeForNormalUserResponseDTO().getSerialNumber());
+                    assertEquals(bike.getBikeType(), responseDTO.getBikeForNormalUserResponseDTO().getBikeType());
+                });
+    }
+
+    @Test
+    public void updateRent_ShouldUpdateDaysOfRentAndMapAndReturnRentResponseDTO_WhenRentIdAndRentRequestDTOIsGiven() {
+
+        Rent rent = new Rent(null, LocalDateTime.of(2023, 11, 23, 10, 0, 0),
+                null, 10, 100.00, bike, user, null);
+        rentRepository.save(rent);
+
+        RentRequestDTO rentRequestDTO = new RentRequestDTO(null, LocalDateTime.of(2023, 11, 23,
+                10, 0, 0), null, 5, null,
+                null, null);
+
+        RentResponseDTO expected = new RentResponseDTO(1L, rentRequestDTO.getRentalStartTime(),
+                rentRequestDTO.getRentalEndTime(), rentRequestDTO.getDaysOfRent(), rentRequestDTO.getBikeForNormalUserResponseDTO(),
+                rentRequestDTO.getUserForNormalUserResponseDTO(), rentRequestDTO.getStationForNormalUserResponseDTO());
+
+        webTestClient.put()
+                .uri("/api/rentals/" + rent.getId())
+                .bodyValue(rentRequestDTO)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(RentResponseDTO.class)
+                .consumeWith(response -> {
+                    RentResponseDTO responseDTO = response.getResponseBody();
+                    assertNotNull(responseDTO);
+                    assertEquals(expected.getDaysOfRent(), responseDTO.getDaysOfRent());
+                });
+
+        rent.setBike(null);
+        rent.setUser(null);
+        rentRepository.save(rent);
+    }
+
+    @Test
+    public void returnBike_ShouldReturnResponseEntityAsStringAndReturnBikeToStation_WhenRentAndStationIdsAreGiven() {
+
+        Rent rent = new Rent(null, LocalDateTime.of(2023, 11, 23, 10, 0, 0),
+                null, 10, 100.00, bike, user, null);
+        rentRepository.save(rent);
+
+        webTestClient.put()
+                .uri(uriBuilder -> uriBuilder
+                        .path("/api/returns/" + rent.getId())
+                        .queryParam("returnChargingStationId", chargingStation.getId())
+                        .build())
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(String.class)
+                .consumeWith(response -> {
+                    String responseMessage = response.getResponseBody();
+                    ChargingStation returnChargingStation =
+                            chargingStationRepository.findById(chargingStation.getId()).orElseThrow(IllegalArgumentException::new);
+                    assertNotNull(responseMessage);
+                    assertEquals("Bike successfully returned.", responseMessage);
+                    assertEquals(chargingStation.getFreeSlots() - 1, returnChargingStation.getFreeSlots());
```

Comment:
Why is there "- 1" here?

Reply:
because this is expected. Expected should be the amount of free slots - 1 (because we return a bike to station so one free slot should be taken)

Reply:
You're right! Thanks for explanation! This makes sense and looks fine :)